keps/sig-cluster-lifecycle/kubeadm/4471-cp-join-kubelet-local-apiserver/README.md

```diff
@@ -0,0 +1,665 @@
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+
+#### Alpha
+
+- Feature implemented behind a feature gate `ControlPlaneKubeletLocalMode`
+- Initial unit and e2e tests completed and enabled
+- [Document the feature gate](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-init/#feature-gates).
+- Document the upcoming breaking change at the release notes.
+
+#### Beta
+
+- Make feature gate to be enabled by default
+- Gather feedback from developers and surveys
+- Make unit and e2e test changes
+- Update the feature gate documentation
+- [Document the new phases](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-join-phase/)
+
+#### GA
+
+- Gather feedback from developers and surveys
+- Update unit tests
+- Remove e2e tests as this will be tested by all existing kubeadm e2e tests
+- Update the feature gate documentation
+- Update the phases documentation
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+- N/A -> Alpha: users can patch their `ClusterConfiguration` in the `kube-system/kubeadm-config` ConfigMap to before calling `kubeadm upgrade apply` This will allow them to enable the `ControlPlaneKubeletLocalMode` in case they wish to join control plane nodes to this cluster. This scenario is anticipated as rare, because usually users maintain a stable control plane with 3 or more members before upgrading it. But it is still plausible and can be documented in the feature gate documentation.
+- Alpha -> Beta: similarly to the previous stage users can modify the `ClusterConfiguration` to disable the feature gate during upgrade. This will allow them to use the "old way", in case they wish to join more control plane nodes to the cluster while the feature gate is enabled by default.
+- Beta -> GA: users could no longer patch the `ClusterConfiguration` to opt-out of the feature and it will be locked to default.
```

Comment:
```suggestion
- Beta -> GA: users could no longer patch the `ClusterConfiguration` to opt-out of the feature and it will be locked to enabled by default.
```